libraries/pipes/implementations/rust/src/lib.rs

```diff
@@ -94,6 +94,17 @@ where
         let msg = PipesMessage::new(Method::ReportAssetCheck, Some(params));
         self.message_channel.write_message(msg)
     }
+
+    pub fn report_custom_message(
+        &mut self,
+        payload: Option<serde_json::Value>,
```

Comment:
Maybe it makes sense to expect a `Value` instead of `Option<Value>` here?

Reply:
Yeah, I think it makes more sense for it to be a `Value` as if you are reporting the message, you likely expect it to be non-null.